docs/reference/api/connectivity/bluetooth/Gap.md

```diff
@@ -1,11 +1,19 @@
-## Gap
+## GAP
 
-[Add description here.]
+The Generic Access Profile is the layer of the stack that handles connectivity tasks. This includes link establishment and termination, advertising and scanning.
 
-### Gap class reference
+Devices with data to publish can use GAP to advertise. They can include the data in the advertisement itself or inside the scan response or leave a peer device to query it after the connection has been established.
 
-[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/v5.7/mbed-os-api-doxy/class_gap.html)
+The other side of the process is the act of scanning, which listens for advertisements, allows you to query the advertisers for more data through a scan request or to connect in order to query the peer device for the data we want.
 
-### Gap example
+Advertising, scanning and connection all have parameters that let you find a compromise between desired power consumption levels and latency and efficiency of these processes.
 
-[Add example here.]
+### GAP class reference
+
+[![View code](https://www.mbed.com/embed/?type=library)](http://os.mbed.com/docs/v5.7/mbed-os-api-doxy/class_gap.html)
+
+### GAP example
+
+Here is an example demonstrating how to use the GAP API to advertise, scan, connect and disconnect and how parameters influence efficiency of these actions.
+
+[![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed-os-examples/code/mbed-os-example-ble-GAP/)](https://os.mbed.com/teams/mbed-os-examples/code/mbed-os-example-ble-GAP)
```

Comment:
Note: I can't get this link to work. Do you know what it should be?

Reply:
the link doesn't work yes because the page hasn't been generated for the example and I don't know what/who triggers the generation of the page based on the new example code

Reply:
Aha. @adbridge Could you please help with this?

Reply:
mbed-os-example-ble-GAP is not yet even a supported example as far as I am aware. It needs to be added to the examples list within mbed-os and also I think @screamerbg needs to do some magic to get it to mirror from github to mbed.com....